config/configgrpc/configgrpc.go

```diff
@@ -261,7 +261,7 @@ func (gcs *ClientConfig) ToClientConn(
 	if err != nil {
 		return nil, err
 	}
-	return grpc.NewClient(gcs.sanitizedEndpoint(), grpcOpts...)
+	return grpc.DialContext(ctx, gcs.sanitizedEndpoint(), grpcOpts...)
```

Comment:
Need to suppress the linter warning https://github.com/open-telemetry/opentelemetry-collector/actions/runs/11598212701/job/32293604408?pr=11575

Reply:
@songy23 do you know where that should be configured